src/core/shape_inference/include/fake_convert_shape_inference.hpp

```diff
@@ -13,16 +13,17 @@ template <class T, class TRShape = result_shape_t<T>>
 std::vector<TRShape> shape_infer(const FakeConvert* op, const std::vector<T>& input_shapes) {
     const auto inputs_count = input_shapes.size();
     const auto has_shifts_input = inputs_count == 3;
-    NODE_VALIDATION_CHECK(op, inputs_count  == 2 || has_shifts_input );
+    NODE_VALIDATION_CHECK(op, inputs_count == 2 || has_shifts_input);
     if (has_shifts_input) {
-        NODE_SHAPE_INFER_CHECK(op, input_shapes, input_shapes[1].compatible(input_shapes[2]),
-                        "FakeConvert scale shape is not compatible with shift shape.");
+        NODE_SHAPE_INFER_CHECK(op,
+                               input_shapes,
+                               input_shapes[1].compatible(input_shapes[2]),
+                               "FakeConvert scale shape is not compatible with shift shape.");
     }
     TRShape output_pshape = input_shapes[0];
-    NODE_VALIDATION_CHECK(
-        op,
-        TRShape::broadcast_merge_into(output_pshape, input_shapes[1], op::AutoBroadcastType::NUMPY),
-        "Argument shapes are inconsistent.");
+    NODE_VALIDATION_CHECK(op,
+                          TRShape::broadcast_merge_into(output_pshape, input_shapes[1], op::AutoBroadcastType::NUMPY),
+                          "Argument shapes are inconsistent.");
     OPENVINO_ASSERT(input_shapes[0].compatible(output_pshape),
```

Comment:
Is it really required if you merge will not check all?
If this assert is required use `NODE_SHAPE_INFER_CHECK` instead.
For future improvements maybe these two checks can be replaced by one

Reply:
The thing is that `broadcast_merge_into` allows for bidirectional broadcast, but broadcasting data_shape to the scale/shift shape shouldn't be allowed. For example bidirectional broadcast is kept in the FakeQuantize shape infer just because of backward compatibility, but it shouldn't be allowed:
https://github.com/openvinotoolkit/openvino/blob/21a90f4e78889863c8e05c82ee54f0d77f50e336/src/core/shape_inference/include/fake_quantize.hpp#L27-L41

Reply:
Yeah, I agree that it's not most optimal way to check if only unidirectional broadcast is possible, but it was quickest to implement. I agree that we should probably consider to replace it with single check. Added NODE_SHAPE_INFER_CHECK as requested